### PR TITLE
Catches the UnicodeError from whois.whois

### PR DESCRIPTION
**Pull request by ohachimOs**
## Handle Unicode errors in WHOIS domain fetch

Adds error handling for invalid domain characters to prevent crashes when processing malformed domains.
[Ticket-os-13354](https://report.ostorlab.co/remediation/tickets/os-13354)
image

### CODE REVIEW

File: tests/whois_domain_agent_test.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 """Unittests for whois_domain agent."""
 
 import datetime
-from typing import List, Any
+from typing import Any, List
```
Comment (nmasdoufi-ol) [inline tests/whois_domain_agent_test.py:4]: use list instead of List.

Comment (benyissa) [inline tests/whois_domain_agent_test.py:4]: would use list

**Review by benyissa — DISMISSED** [review]

File: tests/whois_domain_agent_test.py (hunk 1 above)
Comment (PiranhaSa) [inline tests/whois_domain_agent_test.py:4]: I think instead of List just use the built in list

**Review by PiranhaSa — APPROVED** [review]

File: tests/whois_domain_agent_test.py (hunk 1 above)
Comment (ohachimOs) [reply tests/whois_domain_agent_test.py]: ofc

Comment (ohachimOs) [reply tests/whois_domain_agent_test.py]: :+1:

**Review by 3asm — APPROVED** [review]

**Pull request merged**